docs/adrs/0076-partial-update-patches.md

```diff
@@ -115,24 +118,41 @@ Due to the scope of work required, implementating this pattern would need to be
+
+### Implementation 
+The implementation of this pattern might vary from handler to handler, but the handlers and service that update the MTOShipment model may serve as a good example.
+
+The flow updating a shipment via the ghc-api is sort of confusing. The [UpdateMTOShipmentHandler](https://github.com/transcom/mymove/blob/de9148371427ba348e94439b4acb766f05013797/pkg/handlers/ghcapi/mto_shipment.go#L284) fetches the existing model from the database, creates a new model from the payload, updates _some_ of the fields in the payload model with fields from the existing model, and passes the payload model to the service. Subsequently, the [MTOShipmentUpdater service](https://github.com/transcom/mymove/blob/de9148371427ba348e94439b4acb766f05013797/pkg/services/mto_shipment/mto_shipment_updater.go) fetches the database model again, updates fields in that model based on fields that are not nil in the payload model, and saves the database model (and does other service things, e.g. validation). In this case, both the handler and the service are in some ways responsible for updating a model with information from the payload and the database model.
+With the implementation of this proposal, the handler still fetches the existing model from the database, and it has the option to [update only the fields in that model that are present in the payload](https://github.com/transcom/mymove/blob/762eae697d87f16a7fecd734dc3ec8a832d73e8b/pkg/handlers/ghcapi/internal/payloads/payload_to_model.go#L346). It can then pass this updated model to the service, which will [no longer fetch the existing database model and check to update each of its fields](https://github.com/transcom/mymove/pull/9858/files#diff-61462ff1f1c3c378fa84dac247390a2b0b2cd35d5de3c2ea7a54d3a858f50572); instead the service will simply save the model, in addition to its other service responsibilities.
 
-## Considered Alternatives
-* *Introduce a new struct that serves between the handler and service layers and broadcasts intent*
+## Considered Options
+* **Intermediate Structs** -- introduce a new struct that serves between the handler and service layers and broadcasts intent
+* **Handler Transformation** -- move any logic for converting the payload to a model from the service to the handler
 * *Update model to store information about "presence"*
 * *Pass payload to service*
 * *Do nothing*
 
 ## Decision Outcome
+### Chosen Option: **Handler Transformation** -- move any logic for converting the payload to a model from the service to the handler
+* `+` *Addresses issue of not being able to delete fields with patches*
+* `+` *Relatively simple to implement in existing handlers*
+* `+` *Doesn't overcomplicate service validators*
+* `-` *Handlers responsible for a lot of logic, including fetching db model and updating that fetched struct with payload intelligently*
```

Comment:
**thought**: I think that while this is true, it could be misread. Like, yes, this proposes we move the fetch to the handler, but not directly. Handlers should use fetcher services to get the current version rather than having an inline fetch. I don't know if we need to make that distinction here or not, but just wanted to call that out.